### PR TITLE
4oh4

### PR DESCRIPTION
**Pull request by MatthewJWalls**
## What does this change?

Port of Simon's 404 page code to this project

## Why?

:thinking:

### CODE REVIEW

**Comment by MatthewJWalls** [comment]
Hi, we decided on a way to (in the very near future) provide configuration like the beacon url from the scala to the node app, so I went ahead and did the conversion of the 404 page now so you didn't have to worry about it.

I'm expecting to have a to do a bunch of changes to this since I'm not as good at clientside dev, in particular I think you probably prefer emotion's ```styled``` to ```css``` so everything about my implementation is up for debate

**Comment by SiAdcock** [comment]
As predicted, I think I'm going to suggest using the `styled()` function to co-locate CSS with the component 😄

### Why?

- It prevents CSS from being reused across a bunch of different components
- When developing, it saves us from flipping between stylesheets and component files
- A component is defined by its styles, markup and logic. If markup and logic are in the same place, why not styles also?
- It's consistent with other components
- `styled()` creates a component that can be extended. It's easier to reason about if all components can be extended consistently

**Comment by GHaberis** [comment]
- There's a PR (https://github.com/guardian/guui/pull/50) pending that'll change the import path of image assets.
- As of yesterday we're using `preact-emotion` rather than `react-emotion`.

**Comment by MatthewJWalls** [comment]
@SiAdcock gotcha, will change to used styled and colocate the code
@GHaberis thanks for the heads up, are the preact changes in master? I'll rebase if so

**Comment by GHaberis** [comment]
@MatthewJWalls yep the preact changes are now in master

**Comment by MatthewJWalls** [comment]
@SiAdcock I have converted to use styled for all components

Still need to rebase to use the new stuff, will finish that in the morning

File: src/components/NotFound/index.js [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+import styled from 'react-emotion';
+import { egyptian } from 'pasteup/fonts';
+import { tablet, desktop, leftCol, wide } from 'pasteup/breakpoints';
+import GuardianLogo from './GuardianLogo';
+import NotFoundLogo from './NotFoundLogo';
+import BackToGuardianButton from './BackToGuardianButton';
+
+const beaconUrl = '//beacon.gu-web.net';
+
+const VisuallyHidden = styled('span')({
```
Comment (sndrs) [inline src/components/NotFound/index.js]: we have a helper for this in pasteup

File: src/components/NotFound/index.js [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+    margin: '-1px',
+    overflow: 'hidden',
+    padding: 0,
+    position: 'absolute',
+    width: '1px',
+});
+
+const NotFoundWrapper = styled('div')({
+    content: '',
+    background: '#e7edef',
```
Comment (sndrs) [inline src/components/NotFound/index.js]: colours should come from pasteup really

File: src/components/NotFound/index.js [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+    position: 'fixed',
+    top: 0,
+    right: 0,
+    bottom: 0,
+    left: 0,
+    zIndex: -1,
+    fontFamily: egyptian,
```
Comment (sndrs) [inline src/components/NotFound/index.js]: font are in pasteup too

File: src/components/NotFound/index.js [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+});
+
+const MainContent = styled('div')({
+    boxSizing: 'border-box',
```
Comment (sndrs) [inline src/components/NotFound/index.js]: on by default

File: src/components/NotFound/index.js [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+    lineHeight: '27px',
+
+    [tablet]: {
+        fontSize: '18px',
+        maxWidth: '300px',
+    },
+});
+
+const NeutralLink = styled('a')({
+    borderBottom: '2px solid #aabfc7',
```
Comment (sndrs) [inline src/components/NotFound/index.js]: color again

File: src/components/NotFound/index.js [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+            <NotFoundLogo />
+            <ContentText>
+                <Heading>
+                    Sorry – the page you have requested does not exist
+                </Heading>
+                <BodyCopy>
+                    You may have followed an outdated link, or have mistyped a
+                    URL. If you believe this to be an error, please&nbsp;
+                    <NeutralLink href="https://www.theguardian.com/info/tech-feedback">
+                        {' '}
```
Comment (sndrs) [inline src/components/NotFound/index.js]: are these spaces necessary?

**Comment by sndrs** [comment]
can we call it 404, rather than not found?

**Comment by MatthewJWalls** [comment]
re the use of 404 rather than NotFound, 404 isn't a valid variable name and I'd like to keep the naming of the vars and the files consistent so I settled on NotFound for everything

File: src/components/NotFound/index.js (hunk 6 above)
Comment (MatthewJWalls) [reply src/components/NotFound/index.js]: I just tried removing them and it causes the 'report it' be line broken between the 'report' and 'it' so I'm assuming these are there to force the wording to not be broken up like that, somehow.

File: src/components/NotFound/index.js (hunk 3 above)
Comment (sndrs) [reply src/components/NotFound/index.js]: ha! sorry!

**Comment by sndrs** [comment]
@MatthewJWalls have merged the housekeeping changes to master and fixed the conflicts in your branch – you're up to date with master now

**Comment by MatthewJWalls** [comment]
@sndrs @SiAdcock think I've accounted for the feedback let me know what you think

File: src/pasteup/palette.js [hunk 7]
```diff
@@ -25,7 +25,10 @@ const palette = {
         medium: '#bb3b80',
         dark: '#7d0068',
     },
-    yellow: '#ffe500',
+    yellow: {
+        light: '#ffe500',
```
Comment (sndrs) [inline src/pasteup/palette.js]: this should prbbaly be medium (which is kind of the default)

Comment (sndrs) [reply src/pasteup/palette.js]: sorry mat - the original (now light) should be medium :)

**Pull request merged**